### ISSUE TITLE
Migrate to TypeScript

### ISSUE THREAD

**Issue by queengooborg** [foolip/mdn-bcd-collector#2013]
In https://github.com/mdn/browser-compat-data/issues/16333, I proposed that BCD's internals are migrated over to TypeScript.  I propose that we do the same here, especially for our tools like `update-bcd` and `add-new-bcd`.

**Comment by foolip** [comment]
I think we should migrate `update-bcd` and `add-new-bcd` to the BCD repo, and for the collector to provide input files to those scripts, so maybe we can rewrite them to TypeScript as part of moving them?

I'd be open to migrating everything that remains, but would like to see how it goes in BCD first, since I'm not familiar with TypeScript tooling myself.

**Comment by queengooborg** [comment]
Scripts that depend on BCD's scripts have been migrated to TypeScript now.  I decided to leave the remaining scripts untouched for the time being, and will possibly follow up on that later.

I decided to hold off on moving the scripts to BCD for now.  While doing so will ensure that updates to BCD's infrastructure don't end up suddenly breaking these scripts, and while I'm confident MDN would say yes to transferring repository ownership, I don't think it would be a good idea to make any related decisions until we have heard a decision from Mozilla.  (I do plan to bring the subject up again in a future BCD meeting!)
